cmd/argocd/commands/app.go

```diff
@@ -946,6 +889,90 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	return command
 }
 
+// DifferenceOption struct to store diff options
+type DifferenceOption struct {
+	local         string
+	localRepoRoot string
+	revision      string
+	cluster       *argoappv1.Cluster
+	res           *repoapiclient.ManifestResponse
+}
+
+func findandPrintDiff(app *argoappv1.Application, resources *applicationpkg.ManagedResourcesResponse, argoSettings *settingspkg.Settings, appName string, foundDiffs *bool, diffOptions *DifferenceOption) {
+	liveObjs, err := liveObjects(resources.Items)
+	errors.CheckError(err)
+	items := make([]objKeyLiveTarget, 0)
+	if diffOptions != nil {
```

Comment:
I think instead of nesting everything into a `if diffOptions != nil` branch, we should either a) assume `diffOptions` is always passed non-nil (and ensure this on the caller level) or b) use call by value instead of call by reference. This would remove some complexity.